dln/operator.py

```diff
@@ -250,66 +230,124 @@ def _mini_batch(self, inputs, batch_size=20):
+        **kwargs,
+    ) -> List[str]:
+        if not isinstance(inputs, list):
             inputs = [inputs]
-
-        kwargs.pop("output_space", None)
         generation_options = self.generation_options.copy()
         generation_options.update(**kwargs)
 
-        if self.engine in ("gpt-3.5-turbo", "gpt-35-turbo", "gpt-4", "gpt-4-32k", "gpt-4-0613", "any"):
+        if self.engine in self.CHAT_COMPLETION_MODELS:
             if "return_logprobs" in generation_options:
+                logging.warn(
+                    "return_logprobs is not supported for chat completion models"
+                )
                 del generation_options["return_logprobs"]
 
             if async_generation is True:
                 # async call api, devide to mini batches to avoid call rate limit
                 outputs = []
                 for input_batch in self._mini_batch(inputs, batch_size=10):
                     outputs_batch = asyncio.run(
-                        self.gather_chat_response(input_batch, **generation_options)
+                        self._gather_chat_response(input_batch, **generation_options)
                     )
                     outputs = outputs + outputs_batch
             else:
                 # call api one by one
                 outputs = [
-                    self.get_chat_completion_response(_input, **generation_options)
+                    asyncio.run(
+                        self._aget_chat_completion_response(_input, **generation_options)
+                    )
                     for _input in inputs
                 ]
         else:
-            # devide to mini batches (max batch size = 20 according to openai)
+            # completion_models, devide to mini batches (max batch size = 20 according to openai)
             outputs = []
             for input_batch in self._mini_batch(inputs, batch_size=batch_size):
-                outputs_batch = self.get_completion_response(
+                outputs_batch = self._get_completion_response(
                     input_batch, **generation_options
                 )
                 outputs = outputs + outputs_batch
         return outputs
 
 
-def forward_instantiate(model_name="text-davinci-003", **generation_options):
-    global forward_interpreter
+class VLLM(LLM):
 
-    if forward_interpreter is None:
-        forward_interpreter = GPT(model_name, **generation_options)
-    else:
-        print("Forward interpreter already instantiated.")
-        pass
+    def __init__(self, model_name="text-davinci-003", **generation_options):
+        super().__init__(model_name, **generation_options)
+        self.encoder = instantiate_tokenizer(model_name)
+
+    @_retry_request(min_wait=1, max_wait=1, max_attempts=100)
+    async def _aget_vllm_response(self, input, **kwargs):
+        response = await openai.Completion.acreate(
+            model=self.engine,
+            # prompt=input,
+            prompt = f"<s>[INST] {input} [/INST]",
```

Comment:
That prompt template is specific to llama family I think.

Reply:
Good catch, thanks! This should not be here, but in the templates.

Reply:
Ideally that should be in vLLM.